Sources/Resolve.swift

```diff
@@ -265,7 +265,7 @@ extension DependencyContainer {
     
     //if no definition registered for exact type try to find type-forwarding definition that can resolve the type
     //that will actually happen only when resolving optionals
-    if definitions.filter({ $0.0.type == key.type }).isEmpty {
+    if definitions[type: key.type].filter({ $0.0.type == key.type }).isEmpty {
```

Comment:
would it make sense to use `.first` instead of `.filter`?

Reply:
or `contains` since it returns a `bool` and will stop at first as well

Reply:
Fixed